stdlib/META.bytes

```diff
@@ -0,0 +1,3 @@
+name="bytes"
+version="[distributed with OCaml]"
+description="dummy backward-compatibility package for mutable strings"
```

Comment:
If I understand correctly, this dummy file makes it possible to depend on the `bytes` library (say from inside a Dune file) without having to install anything else apart from the compiler. There are other such compatibility shims which play the same role for other parts of the standard library:

- `META.seq`: https://github.com/ocaml/opam-repository/blob/master/packages/seq/seq.base/opam
- `META.bigarray`: https://github.com/ocaml/opam-repository/blob/master/packages/base-bigarray/base-bigarray.base/opam

Could we consider shipping them as part of the compiler as well? While this may not change much for `opam` users, it does for non-`opam` users (eg Windows) where the current situation means you are forced to "install" these compatibility packages (which are empty) just to get the library names into the environment.

Reply:
Distributing those packages sounds like a good idea, I kind of remember that the `META.bigarray` might require a more subtle handling, but I will have a look.

Reply:
@dra27 correctly clarified that indeed we don't want to ship `META.bigarray` at all because there won't be a `bigarray` package anymore in the future.

Reply:
There's a much longer answer available on this, but TL;DR I (quite strongly) don't think we should not ship `META.bytes` (or any other _back-porting_ `META` files). Cross-referencing https://github.com/ocaml/opam-repository/pull/21333#issuecomment-1153276739, for Dune in particular, this can be resolved in a similar way to `bigarray` - although it can be resolved much more simply because _no project using Jbuilder or Dune should ever have referred to the findlib bytes package_!

For Jbuilder/Dune packages, if a package (incorrectly) refers to the `bytes` package in a `libraries` field, then its opam file should also refer to `base-bytes`, which will pull in ocamlfind. ocamlfind will then be responsible for the `bytes` package (it's a backport package which just happens to live in findlib) and Dune will then be fine. We're doing vast amounts of metadata adjustment in opam-repository with the normal readiness testing for 5.0, so adjusting these will not be a problem. The important point with `base-bytes` is that we don't force releases of these packages - there's an incentive to release corrected versions without a dependency on `base-bytes` because it lifts the findlib dependency, but there's not a requirement to release, because we just amend the dependencies in opam-repository.

Reply:
Ugh - I was quite strongly seeing double as usual! I think we should _not_ ship it - sorry for muddying that in the comment (I've edited it!)